central/testutils/testexport.go

```diff
@@ -0,0 +1,348 @@
+package testutils
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/golang/protobuf/jsonpb"
+	grpcMiddleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	deploymentDataStore "github.com/stackrox/rox/central/deployment/datastore"
+	imageDataStore "github.com/stackrox/rox/central/image/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/random"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/testutils"
+	"github.com/stackrox/rox/pkg/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+const (
+	namespace10pct = "Namespace10%"
+	namepsace90pct = "Namespace90%"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// ExportServicePostgresTestHelper is a utility to help testing the
+// export APIs (takes over the data injection).
+type ExportServicePostgresTestHelper struct {
+	Ctx         context.Context
+	pool        *pgtest.TestPostgres
+	Deployments deploymentDataStore.DataStore
+	Images      imageDataStore.DataStore
+}
+
+// SetupTest prepares the ExportServicePostgresTestHelper struct for testing.
+func (h *ExportServicePostgresTestHelper) SetupTest(tb testing.TB) error {
+	h.Ctx = sac.WithGlobalAccessScopeChecker(
+		context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Deployment, resources.Image),
+		),
+	)
+	h.pool = pgtest.ForT(tb)
+	deploymentStore, err := deploymentDataStore.GetTestPostgresDataStore(tb, h.pool)
+	if err != nil {
+		return err
+	}
+	h.Deployments = deploymentStore
+	h.Images = imageDataStore.GetTestPostgresDataStore(tb, h.pool)
+	return nil
+}
+
+// TearDownTest cleans up the ExportServicePostgresTestHelper resources after testing.
+func (h *ExportServicePostgresTestHelper) TearDownTest(tb testing.TB) {
+	h.pool.Teardown(tb)
+	h.pool.Close()
+}
+
+// getAuthInterceptor returns a GRPC Steam Interceptor that overrides the server context to make sure all calls are authenticated.
+func getAuthInterceptor(ctx context.Context) func(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
```

Comment:
might as well make this a struct method to not have the context as a receiver but just use the property?

Reply:
This is just a helper function for the GRPC streaming service creation.
I'll move the GRPC server creation to `pkg/grpc` so that it is in a central place as suggested. I might inline this function there.